### PR TITLE
Add column type to a query result of InfluxDB

### LINKED ISSUE #6179: Error when running CSV download with InfluxDB as data source
Below is the error log for the Redash application.

Downloading CSV with InfluxDB as the data source causes an error.
The error log shows the following error
Since the test connection succeeds, we believe it is an application anomaly in Redash.

```
server_1            | KeyError: 'type'
server_1            | [2023-07-07 05:01:02,196][PID:9087][ERROR][redash.app] Exception on /api/queries/71/results/2093.csv [GET]
server_1            | Traceback (most recent call last):
server_1            |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1949, in full_dispatch_request
server_1            |     rv = self.dispatch_request()
server_1            |   File "/usr/local/lib/python3.7/site-packages/flask/app.py", line 1935, in dispatch_request
server_1            |     return self.view_functions[rule.endpoint](**req.view_args)
server_1            |   File "/usr/local/lib/python3.7/site-packages/flask_restful/__init__.py", line 458, in wrapper
server_1            |     resp = resource(*args, **kwargs)
server_1            |   File "/usr/local/lib/python3.7/site-packages/flask_login/utils.py", line 261, in decorated_view
server_1            |     return func(*args, **kwargs)
server_1            |   File "/usr/local/lib/python3.7/site-packages/flask/views.py", line 89, in view
server_1            |     return self.dispatch_request(*args, **kwargs)
server_1            |   File "/app/redash/handlers/base.py", line 33, in dispatch_request
server_1            |     return super(BaseResource, self).dispatch_request(*args, **kwargs)
server_1            |   File "/usr/local/lib/python3.7/site-packages/flask_restful/__init__.py", line 573, in dispatch_request
server_1            |     resp = meth(*args, **kwargs)
server_1            |   File "/app/redash/permissions.py", line 71, in decorated
server_1            |     return fn(*args, **kwargs)
server_1            |   File "/app/redash/handlers/query_results.py", line 409, in get
server_1            |     response = response_builders[filetype](query_result)
server_1            |   File "/app/redash/handlers/query_results.py", line 439, in make_csv_response
server_1            |     serialize_query_result_to_dsv(query_result, ","), 200, headers
server_1            |   File "/app/redash/serializers/query_result.py", line 87, in serialize_query_result_to_dsv
server_1            |     fieldnames, special_columns = _get_column_lists(query_data["columns"] or [])
server_1            |   File "/app/redash/serializers/query_result.py", line 68, in _get_column_lists
server_1            |     if col["type"] == col_type:
server_1            | KeyError: 'type'
```
Below is the JSON data when CSV download fails.
(Some parts are excerpted due to the length of the file.)

```
{
  "query_result": {
    "id": 2093,
    "query": "SELECT\n    SUM(d3) / 60\nFROM\n    charger \nWHERE\n    chg_id = '1'\n    AND time >= '2023-04-30T15:00:00Z'\n    AND time < '2023-05-03T15:00:00Z'\nGROUP BY chg_id, time(15m) fill(linear)\nlimit 10;",
    "data": {
      "columns": [{ "name": "time" }, { "name": "sum" }, { "name": "chg_id" }],
      "rows": [
        { "time": "2023-04-30T15:00:00Z", "sum": 0, "chg_id": "1" },
      ]
    }
  }
}
```

Below is the JSON data for a successful CSV download.

```
      "columns": [
        { "name": "id", "friendly_name": "id", "type": "integer" },
        {
          "name": "project_id",
          "friendly_name": "project_id",
          "type": "integer"
        },
        { "name": "name", "friendly_name": "name", "type": "string" },
        { "name": "filters", "friendly_name": "filters", "type": null },
        { "name": "user_id", "friendly_name": "user_id", "type": "integer" },
        {
          "name": "column_names",
          "friendly_name": "column_names",
          "type": null
        },
        {
```
In the normal case, an element is generated for each column,
In the case of failed data, three columns made up one element.
Therefore, the required type element was missing, resulting in a program error.
The following is a part of the program that caused the error.
```
    for col in columns:
        fieldnames.append(col["name"])

        for col_type in special_types.keys():
            if col["type"] == col_type:
                special_columns[col["name"]] = special_types[col_type]
```
This is just a guess, I am guessing, but it seems to me that the SQL results being executed against InfluxDB are special and are not being converted properly in Redash, which is what is causing the problem.

All text translated by Google Translate.

### PR DESCRIPTION
**Pull request by masayuki038**
## What type of PR is this? 

- [x] Bug Fix

## Description
Bug fix for #6179 

I confirmed that it works fine to download these files (below) in a query page of Redash by manual test:
- csv
- tsv
- excel

## How is this tested?

- [x] Unit tests (pytest, jest)
- [x] Manually

## Related Tickets & Documents
Closes #6179

### CODE REVIEW

**Comment by justinclift** [comment]
@masayuki038 Thanks for this PR. :smile:

It looks like it needs some adjustment to the formatting, in order to pass our CI checks:

```
tests/query_runner/test_influx_db.py:1:1: I001 [*] Import block is un-sorted or un-formatted
tests/query_runner/test_influx_db.py:13:5: F401 [*] `redash.query_runner.influx_db.InfluxDB` imported but unused
```

**Comment by masayuki038** [comment]
@justinclift Thanks for your following up. I fixed the unused import issue and pushed the branch again. However, it failed...
```
Run docker-compose logs
  docker-compose logs
  shell: /usr/bin/bash -e {0}
  env:
    NODE_VERSION: 16.[2](https://github.com/getredash/redash/actions/runs/6440340439/job/17489465013?pr=6505#step:10:2)0.1
Couldn't find env file: /home/runner/work/redash/redash/.env
Error: Process completed with exit code 1.
```
Should I add .env to this branch ?

**Comment by guidopetri** [comment]
Huh, that check should not have run. I'll take a look at it later. For now you can just ignore the failing check

File: tests/query_runner/test_influx_db.py
Comment (guidopetri) [inline tests/query_runner/test_influx_db.py:28]: nit: can we rename the tests to be more descriptive? Maybe something like `test_influxdb_result_types_with_rows` and `test_influxdb_result_types_with_no_rows_are_string`?

**Review by guidopetri — APPROVED** [review]
This looks great, thanks so much @masayuki038 ! I left one more comment but it's a nit. I'm good with merging if you don't want to change the names, but I'll wait until tomorrow to merge in case you do :)

**Review by guidopetri — APPROVED** [review]

**Comment by guidopetri** [comment]
Thanks so much @masayuki038 !

**Pull request merged**
